app/src/main/scala/org/alephium/explorer/persistence/model/AppState.scala

```diff
@@ -0,0 +1,51 @@
+// Copyright 2018 The Alephium Authors
+// This file is part of the alephium project.
+//
+// The library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the library. If not, see <http://www.gnu.org/licenses/>.
+
+package org.alephium.explorer.persistence.model
+
+import akka.util.ByteString
+
+import org.alephium.serde
+import org.alephium.serde._
+import org.alephium.util.TimeStamp
+
+final case class AppState(
+    key: String,
+    value: AppState.Value
+)
+object AppState {
+
+  trait Value
```

Comment:
Nit: I personally think this abstraction is over-engineering, as the values of AppState are not correlated. Therefore a pure ByteString might be better. We then let each state to handle encoding/decoding separately. Here, we should really provide a key-value table.

I am fine with this solution, but it's again KISS principle.